collections.js

```diff
@@ -7,7 +7,7 @@ global.List = require("./list");
 global.Set = require("./set");
 global.Map = require("./map");
 global.MultiMap = require("./multi-map");
-global.WeakMap = require("weak-map");
+global.WeakMap = window.WeakMap || require("weak-map");
```

Comment:
No, it will not avoid loads in production or development. In development, it will download the text, but will skip parsing and execution. In production, it will be included in the bundle, so it’s load and parse are unconditional. Script loading is used in production only for progressive loading and sharding.